### ISSUE TITLE
resolve submodule with 'exports' filed failed

### ISSUE THREAD

**Issue by alienzhou** [esm-dev/esm.sh#356]
Currently esm.sh seems not completely exact to resolve the module with the 'exports' filed.

For example,  importing `es/utils/rand.mjs` from element-plus@2.6.6 will return a wrong resolved (besides it has another bug now). And there are some other abnormal cases. 

Would esm.sh consider following the nodejs resolution spec and [algorithm](https://nodejs.org/api/esm.html#resolution-algorithm)?

**Comment by ije** [comment]
The `exports` is totally chaos, i tried to respect the spec but there are a lot of edge cases

**Comment by alienzhou** [comment]
Agree. It's a chaos. I'm using esm.sh in some projects and find some errors with several packages. When skipping the `exports` all things go well.

Is it necessary to provide an option which could ignore the `exports`?

**Comment by ije** [comment]
i'm not sure, there are too many options now... i will look into the `element-plus` package

**Comment by mb21** [comment]
But is there a way to write a package that bundles a submodule?

This currently isn't working for me: https://esm.sh/jsr/@mastrojs/mastro@0.7.3/service-worker?bundle – it has three imports that aren't bundled for some reason?
